lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+        its('Client.Version') { should cmp >= '1.12'}
+      end
+
+      describe docker.inspect('e0825339329b') do
+        its(%w(HostConfig Privileged)) { should eq false }
+        its(%w(HostConfig Privileged)) { should_not eq true }
+      end
+
+      docker.containers.ids.each do |id|
+        describe docker.inspect(id) do
+          its(%w(HostConfig Privileged)) { should eq false }
+          its(%w(HostConfig Privileged)) { should_not eq true }
+        end
+      end
+
+      describe file(docker.path) do
+        it { should exist }
+        it { should be_file }
+        it { should be_owned_by 'root' }
+        it { should be_grouped_into 'root' }
+      end
+
+      describe file(docker.socket) do
+        it { should exist }
+        it { should be_file }
+        it { should be_owned_by 'root' }
+        it { should be_grouped_into 'root' }
+      end
+    "
+
+    def containers
+      DockerContainerFilter.new(parse_containers)
+    end
+
+    def images
+      DockerImageFilter.new(parse_images)
+    end
+
+    # returns the path of the docker service
+    # only works for Debian 8+, Ubuntu 16.04+ and CentOS 7+ where systemd is installed
+    def path
+      # TODO: use exception, once https://github.com/chef/inspec/issues/1205 is fixed
+      return nil unless is_systemd_system
+
+      cmd = inspec.command('systemctl show -p FragmentPath docker.service')
+      return nil if cmd.exit_status.to_i.nonzero?
+
+      # parse data
+      params = parse_systemd_values(cmd.stdout.chomp)
+
+      # return the value
+      params['FragmentPath']
+    end
+
+    # returns the docker socket
+    # only works for Debian 8+, Ubuntu 16.04+ and CentOS 7+ where systemd is installed
+    def socket
+      # TODO: use exception, once https://github.com/chef/inspec/issues/1205 is fixed
+      return nil unless is_systemd_system
+
+      cmd = inspec.command('systemctl show -p FragmentPath docker.socket')
+      return nil if cmd.exit_status.to_i.nonzero?
+
+      # parse data
+      params = parse_systemd_values(cmd.stdout.chomp)
+
+      # return the value
+      params['FragmentPath']
+    end
+
+    def to_s
+      'Docker Host'
+    end
+
+    # returns the json output of `docker inspect`
+    def inspect(id)
+      return @inspect if defined?(@inspect)
+      data = JSON.parse(inspec.command("docker inspect #{id}").stdout)
+      # we'll return the first value only
+      data = data[0] if data.is_a?(Array)
+      @inspect = Hashie::Mash.new(data)
+    end
+
+    # returns the json object of the info
+    def version
+      return @version if defined?(@version)
+      data = JSON.parse(inspec.command('docker version --format \'{{ json . }}\'').stdout)
+      @version = Hashie::Mash.new(data)
+    end
+
+    private
+
+    def parse_containers
+      raw_containers = inspec.command('docker ps -a --no-trunc --format \'{{ json . }}\'').stdout
+      ps = []
+      # since docker is not outputting valid json, we need to parse each row
```

Comment:
LOL - thanks for nothing, docker! Just realized they output individual lines of valid JSON, but not a full JSON structure. Sigh.